In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_excel("/Users/ganesha/Downloads/2LMarketBasket.xlsx")

In [3]:
df.shape

(10931, 44)

In [4]:
df.rename(columns = {'2L+ TKI Target':'2Lplus',
                     'LTD Reach - Live':'LiveRepTouch',
                     'LTD Reach - All Reach Methods':'AnyTouch',
                     'LTD QINLOCK Rx (HCP)' : 'WroteQinlock',
                     'April Marketing Campaign - Email Open Count':'AprilEmailOpens'},inplace = True)

In [5]:
list(df)

['NPI',
 'HCP ID',
 'First Name',
 'Last Name',
 'Specialty',
 'Sales Tier',
 'Facility Tier',
 '2Lplus',
 'AnyTouch',
 'LiveRepTouch',
 'AprilEmailOpens',
 'Total GIST Patients',
 'Total Treated GIST Patients',
 'Total 3L+ GIST Patients',
 'Total Ayva GIST Patients',
 'Komodo Clinical Experience Score',
 'Komodo Clinical Volume Score',
 'GLEEVEC Units - 52 Week',
 'GLEEVEC Units - 26 Week',
 'Sutent Units - 52 Week',
 'Sutent Units - 26 Week',
 'Stivarga Units - 52 Week',
 'Stivarga Units - 26 Week',
 'Ayva Units - 52 Week',
 'Ayva Units - 26 Week',
 'Other TKI Units - 52 Week',
 'Other TKI Units - 26 Week',
 '2L+ TKI Units - 52 Week',
 '2L+ TKI Units - 26 Week',
 'Field Reported 2L+',
 'WroteQinlock',
 'Region',
 'Territory',
 'Facility ID',
 'Facility Name',
 'Facility Address',
 'Facility Address 2',
 'Facility City',
 'Facility State',
 'Facility Zip',
 'Facility Phone',
 'Facility Fax',
 'Network/Parent',
 'Facility Segment']

In [6]:
pd.pivot_table(df,index=["2Lplus","Region"])

2L+ TKI Units - 26 Week  2L+ TKI Units - 52 Week  \
2Lplus Region                                                          
No     Central                     0.132107                 0.244147   
       Great Lakes                 0.209841                 0.399421   
       Northeast                   0.023739                 0.051929   
       South                       0.161572                 0.269287   
       Southeast                   0.082982                 0.205345   
       West                        0.219638                 0.400517   
Yes    Central                     2.041885                 3.921466   
       Great Lakes                 2.004902                 4.004902   
       Northeast                   2.657754                 4.791444   
       South                       1.932990                 3.886598   
       Southeast                   1.862559                 3.530806   
       West                        1.322581                 2.562724   

                    AnyTouch  AprilEmailOpens  Ayva Units - 26 Week  \
2Lplus Region                                                         
No     Central      1.774658         3.540000              0.005017   
       Great Lakes  2.514036         3.204724              0.000000   
       Northeast    1.439294         3.120846              0.000000   
       South        2.276278         3.047210              0.000000   
       Southeast    3.586261         3.040590              0.000000   
       West         1.839329         3.388889              0.002584   
Yes    Central      5.138393         3.181818              0.026178   
       Great Lakes  6.476596         3.638298              0.014706   
       Northeast    4.676190         2.808511              0.000000   
       South        6.131455         3.409091              0.041237   
       Southeast    8.902834         2.756098              0.052133   
       West         4.927978         3.016129              0.050179   

                    Ayva Units - 52 Week  Facility Zip  Field Reported 2L+  \
2Lplus Region                                                                
No     Central                  0.005017  64390.398630            0.000000   
       Great Lakes              0.000000  34963.549595            0.000000   
       Northeast                0.000000  10198.255519            0.000000   
       South                    0.000000  70357.611506            0.000000   
       Southeast                0.000000  31826.834819            0.000000   
       West                     0.002584  92779.757794            0.000000   
Yes    Central                  0.047120  60466.223214            1.489247   
       Great Lakes              0.014706  34494.000000            1.656805   
       Northeast                0.016043   9371.661905            1.333333   
       South                    0.103093  71173.211268            1.356688   
       Southeast                0.061611  31603.441296            1.241935   
       West                     0.064516  93192.650970            1.484746   

                    GLEEVEC Units - 26 Week  GLEEVEC Units - 52 Week  \
2Lplus Region                                                          
No     Central                     2.070234                 4.133779   
       Great Lakes                 2.043415                 4.046310   
       Northeast                   2.243323                 4.379822   
       South                       2.314410                 4.499272   
       Southeast                   2.032349                 3.893108   
       West                        2.240310                 4.447028   
Yes    Central                     4.062827                 8.272251   
       Great Lakes                 5.171569                10.421569   
       Northeast                   6.759358                13.572193   
       South                       4.577320                 9.195876   
       Southeast                   5.199052                

In [7]:
pd.pivot_table(df,index=["2Lplus","Region"],values=["AprilEmailOpens"],aggfunc=[np.sum])

sum
                   AprilEmailOpens
2Lplus Region                     
No     Central               885.0
       Great Lakes           814.0
       Northeast            1033.0
       South                 710.0
       Southeast             824.0
       West                  671.0
Yes    Central               140.0
       Great Lakes           171.0
       Northeast             132.0
       South                 150.0
       Southeast             113.0
       West                  187.0

In [8]:
pd.pivot_table(df,index=["2Lplus","Region"],values=["Total 3L+ GIST Patients","AprilEmailOpens"],aggfunc=[np.sum])

sum                        
                   AprilEmailOpens Total 3L+ GIST Patients
2Lplus Region                                             
No     Central               885.0                     6.0
       Great Lakes           814.0                    10.0
       Northeast            1033.0                     1.0
       South                 710.0                     7.0
       Southeast             824.0                     9.0
       West                  671.0                     8.0
Yes    Central               140.0                    31.0
       Great Lakes           171.0                    44.0
       Northeast             132.0                    53.0
       South                 150.0                    45.0
       Southeast             113.0                    49.0
       West                  187.0                    34.0

In [9]:
pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients"],
               aggfunc=[np.sum])

sum                            
                   Total 3L+ GIST Patients Total Treated GIST Patients
2Lplus Region                                                         
No     Central                         6.0                       418.0
       Great Lakes                    10.0                       499.0
       Northeast                       1.0                       502.0
       South                           7.0                       530.0
       Southeast                       9.0                       488.0
       West                            8.0                       327.0
Yes    Central                        31.0                       382.0
       Great Lakes                    44.0                       472.0
       Northeast                      53.0                       610.0
       South                          45.0                       431.0
       Southeast                      49.0                       524.0
       West                           34.0                       522.0

In [10]:
df["2Lplus"].value_counts()

No     9441
Yes    1490
Name: 2Lplus, dtype: int64

+ (41) 3L Patients among 9441 HCPs **THAT ARE NOT** 2L plus
+ (256) 3L Patients among 1490 HCPs **THAT ARE** 2L plus

In [11]:
pd.pivot_table(df,index=["2Lplus"],
               values=["Total 3L+ GIST Patients"],
               columns=["Region"],
               aggfunc=[np.sum])

sum                                            
       Total 3L+ GIST Patients                                            
Region                 Central Great Lakes Northeast South Southeast  West
2Lplus                                                                    
No                         6.0        10.0       1.0   7.0       9.0   8.0
Yes                       31.0        44.0      53.0  45.0      49.0  34.0

In [12]:
table = pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients","LiveRepTouch","AnyTouch"],
               aggfunc=[np.sum])
table

sum                                       \
                   AnyTouch LiveRepTouch Total 3L+ GIST Patients   
2Lplus Region                                                      
No     Central         2591          652                     6.0   
       Great Lakes     4030          801                    10.0   
       Northeast       2608          234                     1.0   
       South           3205          492                     7.0   
       Southeast       6839         1159                     9.0   
       West            2301          544                     8.0   
Yes    Central         1151          408                    31.0   
       Great Lakes     1522          340                    44.0   
       Northeast        982          159                    53.0   
       South           1306          262                    45.0   
       Southeast       2199          550                    49.0   
       West            1779          477                    34.0   

                                                
                   Total Treated GIST Patients  
2Lplus Region                                   
No     Central                           418.0  
       Great Lakes                       499.0  
       Northeast                         502.0  
       South                             530.0  
       Southeast                         488.0  
       West                              327.0  
Yes    Central                           382.0  
       Great Lakes                       472.0  
       Northeast                         610.0  
       South                             431.0  
       Southeast                         524.0  
       West                              522.0

In [13]:
table = pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","LiveRepTouch","AnyTouch","AprilEmailOpens"],
               aggfunc=[np.sum])
table

sum                               \
                   AnyTouch AprilEmailOpens LiveRepTouch   
2Lplus Region                                              
No     Central         2591           885.0          652   
       Great Lakes     4030           814.0          801   
       Northeast       2608          1033.0          234   
       South           3205           710.0          492   
       Southeast       6839           824.0         1159   
       West            2301           671.0          544   
Yes    Central         1151           140.0          408   
       Great Lakes     1522           171.0          340   
       Northeast        982           132.0          159   
       South           1306           150.0          262   
       Southeast       2199           113.0          550   
       West            1779           187.0          477   

                                            
                   Total 3L+ GIST Patients  
2Lplus Region                               
No     Central                         6.0  
       Great Lakes                    10.0  
       Northeast                       1.0  
       South                           7.0  
       Southeast                       9.0  
       West                            8.0  
Yes    Central                        31.0  
       Great Lakes                    44.0  
       Northeast                      53.0  
       South                          45.0  
       Southeast                      49.0  
       West                           34.0

In [14]:
list(table)

[('sum', 'AnyTouch'),
 ('sum', 'AprilEmailOpens'),
 ('sum', 'LiveRepTouch'),
 ('sum', 'Total 3L+ GIST Patients')]

In [15]:
table

sum                               \
                   AnyTouch AprilEmailOpens LiveRepTouch   
2Lplus Region                                              
No     Central         2591           885.0          652   
       Great Lakes     4030           814.0          801   
       Northeast       2608          1033.0          234   
       South           3205           710.0          492   
       Southeast       6839           824.0         1159   
       West            2301           671.0          544   
Yes    Central         1151           140.0          408   
       Great Lakes     1522           171.0          340   
       Northeast        982           132.0          159   
       South           1306           150.0          262   
       Southeast       2199           113.0          550   
       West            1779           187.0          477   

                                            
                   Total 3L+ GIST Patients  
2Lplus Region                               
No     Central                         6.0  
       Great Lakes                    10.0  
       Northeast                       1.0  
       South                           7.0  
       Southeast                       9.0  
       West                            8.0  
Yes    Central                        31.0  
       Great Lakes                    44.0  
       Northeast                      53.0  
       South                          45.0  
       Southeast                      49.0  
       West                           34.0

In [16]:
settings = pd.pivot_table(df,index=["2Lplus","Region"],
                values=["Total 3L+ GIST Patients"],
                columns=["Facility Segment"],
                aggfunc=[len])
settings

len                                 \
                   Total 3L+ GIST Patients                                  
Facility Segment                  AMC Flag IDN Flag Independent Hospitals   
2Lplus Region                                                               
No     Central                       271.0    802.0                  31.0   
       Great Lakes                   414.0    785.0                  39.0   
       Northeast                     753.0    750.0                  55.0   
       South                         250.0    663.0                  35.0   
       Southeast                     313.0    803.0                  68.0   
       West                          265.0    548.0                  87.0   
Yes    Central                        57.0    104.0                   7.0   
       Great Lakes                    57.0    111.0                   4.0   
       Northeast                     114.0     65.0                   6.0   
       South                          44.0     98.0                   7.0   
       Southeast                      49.0    101.0                   9.0   
       West                           76.0    160.0                  23.0   

                                             
                                             
Facility Segment   Physician Group Practice  
2Lplus Region                                
No     Central                        282.0  
       Great Lakes                    216.0  
       Northeast                      210.0  
       South                          390.0  
       Southeast                      552.0  
       West                           228.0  
Yes    Central                         49.0  
       Great Lakes                     44.0  
       Northeast                       19.0  
       South                           47.0  
       Southeast                       66.0  
       West                            78.0

In [17]:
secondlineplus = df[df["2Lplus"] == "Yes"]
secondlineplus.shape

(1490, 44)

In [18]:
scripts = pd.pivot_table(secondlineplus,index=["WroteQinlock","Region"],
                values=["Total 3L+ GIST Patients"],
                columns=["Facility Segment"],                         
                aggfunc=[len])
scripts

len           \
                         Total 3L+ GIST Patients            
Facility Segment                        AMC Flag IDN Flag   
WroteQinlock Region                                         
No           Central                        34.0     77.0   
             Great Lakes                    33.0     91.0   
             Northeast                      65.0     57.0   
             South                          26.0     82.0   
             Southeast                      30.0     79.0   
             West                           59.0    129.0   
Yes          Central                        23.0     27.0   
             Great Lakes                    24.0     20.0   
             Northeast                      49.0      8.0   
             South                          18.0     16.0   
             Southeast                      19.0     22.0   
             West                           17.0     31.0   

                                                                         
                                                                         
Facility Segment         Independent Hospitals Physician Group Practice  
WroteQinlock Region                                                      
No           Central                       6.0                     40.0  
             Great Lakes                   2.0                     33.0  
             Northeast                     5.0                     13.0  
             South                         6.0                     36.0  
             Southeast                     5.0                     50.0  
             West                         19.0                     71.0  
Yes          Central                       1.0                      9.0  
             Great Lakes                   2.0                     11.0  
             Northeast                     1.0                      6.0  
             South                         1.0                     11.0  
             Southeast                     4.0                     16.0  
             West                          4.0                      7.0

In [19]:
thirdLnoQ = df[(df["2Lplus"] == "Yes")
    & (df["Total 3L+ GIST Patients"] > 0)
    & (df["WroteQinlock"]=="No")]
thirdLnoQ.shape

(132, 44)

In [20]:
thirdLsettings = pd.pivot_table(thirdLnoQ,index=["Region"],
                values=["Total 3L+ GIST Patients"],
                columns=["Facility Segment"],                         
                aggfunc=[len])
thirdLsettings

len                                 \
                 Total 3L+ GIST Patients                                  
Facility Segment                AMC Flag IDN Flag Independent Hospitals   
Region                                                                    
Central                              2.0     10.0                   NaN   
Great Lakes                          6.0     13.0                   NaN   
Northeast                           12.0     12.0                   1.0   
South                                3.0      9.0                   1.0   
Southeast                            5.0     16.0                   NaN   
West                                 4.0      5.0                   2.0   

                                           
                                           
Facility Segment Physician Group Practice  
Region                                     
Central                               2.0  
Great Lakes                           3.0  
Northeast                             1.0  
South                                 3.0  
Southeast                             6.0  
West                                 10.0

In [21]:
thirdLnoQidn = df[(df["2Lplus"] == "Yes")
    & (df["Total 3L+ GIST Patients"] > 0)
    & (df["WroteQinlock"]=="No")
    & (df["Facility Segment"]=="IDN Flag")]
thirdLnoQidn.shape

(65, 44)

In [22]:
thirdLnoQidn

NPI      HCP ID  First Name    Last Name              Specialty  \
707    1568453298  DA01010233       MALIK     BANDEALY  HEMATOLOGY & ONCOLOGY   
710    1497189468  DA01010234     ALLISON       HARDIN  HEMATOLOGY & ONCOLOGY   
711    1588655351  DA01001549       KELLY      KONESKO     NURSE PRACTITIONER   
716    1053302729  DA01004638     MICHAEL      MULLANE  HEMATOLOGY & ONCOLOGY   
1247   1942305677  DA01002775      TRAVIS      KOENEKE  HEMATOLOGY & ONCOLOGY   
1462   1982665337  DA01002022      THOMAS      BUROKER  HEMATOLOGY & ONCOLOGY   
1463   1477782522  DA01004524     PRATIMA    CHALASANI      INTERNAL MEDICINE   
1465   1356538003  DA01001590      BROOKE      GILLETT  HEMATOLOGY & ONCOLOGY   
1466   1073501607  DA01005750      ALFRED        GRECO  HEMATOLOGY & ONCOLOGY   
1475   1801802996  DA01007424       VAMSI    VASIREDDY  HEMATOLOGY & ONCOLOGY   
1708   1114977246  DA01003528   ELIZABETH      GALANTE     NURSE PRACTITIONER   
2036   1275526089  DA01000522       IRFAN      FIRDAUS  HEMATOLOGY & ONCOLOGY   
2055   1023246949  DA01004571        AMIE      JACKSON  HEMATOLOGY & ONCOLOGY   
2064   1164859278  DA01010474     SUMAIRA        SHAFI  HEMATOLOGY & ONCOLOGY   
2066   1669867966  DA01011881        KATY       TICONA  HEMATOLOGY & ONCOLOGY   
2372   1619228483  DA01011763       HOSAM        HAKIM  HEMATOLOGY & ONCOLOGY   
2699   1861525560  DA01011742        AHAD        SADIQ  HEMATOLOGY & ONCOLOGY   
2964   1467693895  DA01003851       ZEINA      EL AMIL      ONCOLOGY, MEDICAL   
2969   1831394287  DA01001906        JOHN        JAKOB  HEMATOLOGY & ONCOLOGY   
2987   1952303471  DA01007924     GREGORY         LONG  HEMATOLOGY & ONCOLOGY   
3248   1598078990  DA01004880      FAYSAL       HAROUN      INTERNAL MEDICINE   
3249   1700903242  DA01007947     WILLIAM        IRVIN  HEMATOLOGY & ONCOLOGY   
3250   1396256012  DA01011321       SUNNY          KIM  HEMATOLOGY & ONCOLOGY   
3537   1154494466  DA01003086      ANDREA      BULLOCK      INTERNAL MEDICINE   
3576   1982831400  DA01010596     JESSICA      ZERILLO  HEMATOLOGY & ONCOLOGY   
3935   1790881803  DA01000470     SUSANNA         HONG      ONCOLOGY, MEDICAL   
4154   1659383669  DA01001727        YUMI          KIM  HEMATOLOGY & ONCOLOGY   
4158   1619351889  DA01011851  JACQUELINE      BELIZAR    PHYSICIAN ASSISTANT   
4163   1134243595  DA01003225       MOHAN        PREET      INTERNAL MEDICINE   
4683   1134125156  DA01007005       KEVIN     CALLAHAN  HEMATOLOGY & ONCOLOGY   
...           ...         ...         ...          ...                    ...   
5569   1669414652  DA01004552     MICHAEL       KASPER  HEMATOLOGY & ONCOLOGY   
5571   1770506297  DA01001172       JESSE     MEDELLIN      ONCOLOGY, MEDICAL   
5811   1639498314  DA01004841     DAI CHU          LUU      INTERNAL MEDICINE   
5820   1659319028  DA01011326        AMIR       FARIDI  HEMATOLOGY & ONCOLOGY   
5830   1588715916  DA01002449       SUNIL        PATEL  HEMATOLOGY & ONCOLOGY   
6168   1851512842  DA01004021       JULIE         ROWE  HEMATOLOGY & ONCOLOGY   
6172   1891947735  DA01002465       PUTAO          CEN      ONCOLOGY, MEDICAL   
6207   1073655965  DA01001985     MONISHA        SINGH  HEMATOLOGY & ONCOLOGY   
6435   1245395573  DA01001262     MATTHEW     MCELVEEN  HEMATOLOGY & ONCOLOGY   
7182   1770769234  DA01002799       SIMBO      ADULOJU  HEMATOLOGY & ONCOLOGY   
7184   1225065766  DA01007189      SHAHIN    CHOWDHURY      ONCOLOGY, MEDICAL   
7209   1073092334  DA01009736     KATELYN    GILLESPIE     NURSE PRACTITIONER   
7212   1740242932  DA01002377        DEAN       KIRKEL  HEMATOLOGY & ONCOLOGY   
7215   1588055891  DA01008772      ANDREA       LIATIS    PHYSICIAN ASSISTANT   
7220   1376520148  DA01007601       DAVID      SQUIRES      ONCOLOGY, MEDICAL   
7603   1659582369  DA01003392       SUSAN      BOSTICK      ONCOLOGY, MEDICAL   
7604   1780663880  DA01006514        MANH         DANG  HEMATOLOGY & ONCOLOGY   
7606   1861475170  DA01005779     RANJITH  DI

In [23]:
idnOpportunity = pd.pivot_table(thirdLnoQidn,index=["Network/Parent"],
                values=["Total 3L+ GIST Patients"],
                columns=["Region"],                        
                aggfunc=[len],fill_value=0)
idnOpportunity
#idnOpportunity.to_csv("/home/jupyter/src/dataviz/podcastcount/idnOpportunity.csv")

len  \
                                                   Total 3L+ GIST Patients   
Region                                                             Central   
Network/Parent                                                               
"QUEEN'S HEALTH SYSTEMS"                                                 0   
"SAINT JOSEPH'S CANDLER HEALTH SYSTEM"                                   0   
ALBERT EINSTEIN HEALTHCARE NETWORK                                       0   
ALLEGHENY HEALTH NETWORK                                                 0   
ASCENSION HEALTH                                                         0   
AURORA HEALTH CARE INC                                                   2   
BETH ISRAEL HEALTHCARE                                                   0   
CANCER TREATMENT CENTERS OF AMERICA                                      0   
CARLE FOUNDATION                                                         2   
CAROLINAS HEALTHCARE SYSTEM                                              0   
CATHOLIC HEALTH INITIATIVES                                              0   
CATHOLIC HEALTH SERVICES OF LONG ISLAND                                  0   
COOPER HEALTH SYSTEM                                                     0   
COXHEALTH                                                                1   
DCH HEALTH SYSTEM                                                        0   
HCA HOLDINGS INC                                                         0   
HOSPITAL SISTERS HEALTH SYSTEM                                           1   
HOUSTON METHODIST                                                        0   
HUNTSVILLE HOSPITAL HEALTH SYSTEM                                        0   
IASIS HEALTHCARE LLC                                                     0   
KALISPELL REGIONAL HEALTHCARE                                            0   
LIFEPOINT HOSPITALS INC                                                  0   
MEMORIAL HERMANN HEALTHCARE SYSTEM                                       0   
MERCY HEALTH MINISTRY SUPPORT SERVICES (FORMERL...                       0   
MERCY HEALTH NETWORK                                                     1   
MIDDLESEX HEALTH SYSTEM                                                  0   
NORTH SHORE-LONG ISLAND JEWISH HEALTH SYSTEM                             0   
NORTHSIDE HOSPITAL HEALTHCARE SYSTEM                                     0   
ROCHESTER REGIONAL HEALTH SYSTEM (AKA VIAHEALTH)                         0   
SAINT TAMMANY PARISH HOSPITAL (SYSTEM)                                   0   
STORMONT-VAIL HEALTHCARE                                                 1   
SUMMA HEALTH SYSTEM                                                      0   
SUTTER HEALTH                                                            0   
TEXAS HEALTH RESOURCES                                                   0   
UNIVERSAL HEALTH SERVICES                                                0   
UNIVERSITY HEALTH CARE SYSTEM                                            0   
WELLSTAR HEALTH SYSTEM                                                   0   
WEST VIRGINIA UNITED HEALTH SYSTEM                                       0   

                                                                          \
                                                                           
Region                                             Great Lakes Northeast   
Network/Parent                                                             
"QUEEN'S HEALTH SYSTEMS"                                     0         0   
"SAINT JOSEPH'S CANDLER HEALTH SYSTEM"                       0         0   
ALBERT EINSTEIN HEALTHCARE NETWORK                           0         1   
ALLEGHENY HEALTH NETWORK                                     1         0   
ASCENSION HEALTH                                             0         0   
AURORA HEALTH CARE INC                                       0         0   
BETH ISRAEL HEALTHCARE            

# prep for mapping

In [24]:
#the untapped opportunities list
thirdLnoQ

NPI      HCP ID  First Name    Last Name              Specialty  \
299    1265482012  DA01000332        ANNE      MARCEAU     NURSE PRACTITIONER   
314    1942310719  DA01009239   ALEXANDER     URQUHART  HEMATOLOGY & ONCOLOGY   
707    1568453298  DA01010233       MALIK     BANDEALY  HEMATOLOGY & ONCOLOGY   
710    1497189468  DA01010234     ALLISON       HARDIN  HEMATOLOGY & ONCOLOGY   
711    1588655351  DA01001549       KELLY      KONESKO     NURSE PRACTITIONER   
715    1437239258  DA01000333    MICHELLE        MOREL    PHYSICIAN ASSISTANT   
716    1053302729  DA01004638     MICHAEL      MULLANE  HEMATOLOGY & ONCOLOGY   
1040   1104204510  DA01008011        PAUL       HAMPEL  HEMATOLOGY & ONCOLOGY   
1247   1942305677  DA01002775      TRAVIS      KOENEKE  HEMATOLOGY & ONCOLOGY   
1462   1982665337  DA01002022      THOMAS      BUROKER  HEMATOLOGY & ONCOLOGY   
1463   1477782522  DA01004524     PRATIMA    CHALASANI      INTERNAL MEDICINE   
1465   1356538003  DA01001590      BROOKE      GILLETT  HEMATOLOGY & ONCOLOGY   
1466   1073501607  DA01005750      ALFRED        GRECO  HEMATOLOGY & ONCOLOGY   
1470   1558429100  DA01000052    MOHAMMED       MILHEM  HEMATOLOGY & ONCOLOGY   
1475   1801802996  DA01007424       VAMSI    VASIREDDY  HEMATOLOGY & ONCOLOGY   
1684   1699907881  DA01004651       TAMMY      BECKMAN     NURSE PRACTITIONER   
1686   1063617710  DA01004075   GEORGANNE     JAMBETER     NURSE PRACTITIONER   
1704   1114994373  DA01007073     RUSSELL       DELUCA  HEMATOLOGY & ONCOLOGY   
1708   1114977246  DA01003528   ELIZABETH      GALANTE     NURSE PRACTITIONER   
1712   1922510742  DA01011136    DHARABEN        PATEL    PHYSICIAN ASSISTANT   
2036   1275526089  DA01000522       IRFAN      FIRDAUS  HEMATOLOGY & ONCOLOGY   
2044   1780632828  DA01001537  TARALKUMAR        PATEL  HEMATOLOGY & ONCOLOGY   
2055   1023246949  DA01004571        AMIE      JACKSON  HEMATOLOGY & ONCOLOGY   
2064   1164859278  DA01010474     SUMAIRA        SHAFI  HEMATOLOGY & ONCOLOGY   
2066   1669867966  DA01011881        KATY       TICONA  HEMATOLOGY & ONCOLOGY   
2372   1619228483  DA01011763       HOSAM        HAKIM  HEMATOLOGY & ONCOLOGY   
2683   1295735058  DA01000905       RUEME     BIRHIRAY  HEMATOLOGY & ONCOLOGY   
2699   1861525560  DA01011742        AHAD        SADIQ  HEMATOLOGY & ONCOLOGY   
2963   1285620468  DA01000430     RICHARD        CHANG  HEMATOLOGY & ONCOLOGY   
2964   1467693895  DA01003851       ZEINA      EL AMIL      ONCOLOGY, MEDICAL   
...           ...         ...         ...          ...                    ...   
7915   1811949308  DA01001435     WILLIAM     MITCHELL  HEMATOLOGY & ONCOLOGY   
8526   1902810500  DA01007117      LESTER    ROBERTSON  HEMATOLOGY & ONCOLOGY   
8536   1477804649  DA01004924     ALLISON      HAWKINS    PHYSICIAN ASSISTANT   
8537   1154312783  DA01008882     VINICIO    HERNANDEZ  HEMATOLOGY & ONCOLOGY   
8540   1194783027  DA01004394       DAVID     MOLTHROP  HEMATOLOGY & ONCOLOGY   
9013   1588616858  DA01004037        LEAH        CLARK     NURSE PRACTITIONER   
9020   1508087297  DA01003161       DAVID         WENK      ONCOLOGY, MEDICAL   
9037   1417153701  DA01003970         BEN          YAN      ONCOLOGY, MEDICAL   
9326   1598758062  DA01000534       BORIS  BAGDASARIAN  HEMATOLOGY & ONCOLOGY   
9333   1114977071  DA01006088      YOURAM       NASSIR      ONCOLOGY, MEDICAL   
9344   1346849007  DA01011443     JANESSA       HAWARA  HEMATOLOGY & ONCOLOGY   
9347   1568816262  DA01008018   ALEXANDER       MARTIN  HEMATOLOGY & ONCOLOGY   
9587   1386671907  DA01001220  CHUNG-TSEN        HSUEH      ONCOLOGY, MEDICAL   
9644   1831458835  DA01007976     NATASHA         GARG  HEMATOLOGY & ONCOLOGY   
9790   1144343005  DA01003026      SACHIN        GUPTA  HEMATOLOGY & ONCOLOGY   
9792   1154541423  DA01002964       NITIN        JOSHI  HEMATOLOGY & ONCOLOGY   
9796   1417927054  DA01000088        RAVI          RAO  HEMATOLOGY & ONCOLOGY   
9797   1710947858  DA01000092   SIVAKUMAR    

In [25]:
#thirdLnoQ["LiveRepTouch"].groupby(df["Facility State"]).count().to_csv("/Users/ganesha/Downloads/CalledOnTargets.csv")
ThirdLineCalledOn = pd.read_csv("/Users/ganesha/Downloads/CalledOnTargets.csv")

In [26]:
ThirdLineCalledOn

State  Patients
0     AL         5
1     AR         2
2     CA        16
3     CO         1
4     CT         1
5     DC         2
6     DE         1
7     FL        10
8     GA         8
9     HI         2
10    IA         2
11    ID         1
12    IL         3
13    IN         2
14    KS         1
15    LA         3
16    MA         2
17    MD         4
18    MI         2
19    MN         1
20    MO         1
21    MS         2
22    MT         1
23    NC         4
24    NH         1
25    NJ         4
26    NY        17
27    OH         6
28    PA         4
29    TN         1
30    TX        10
31    UT         1
32    VA         3
33    WA         2
34    WI         4
35    WV         2

In [27]:
url = (
    "https://raw.githubusercontent.com/python-visualization/folium/master/examples/data"
)
state_geo = f"{url}/us-states.json"

import folium

# initialize the map and store it in a m object
m = folium.Map(location=[40, -95], zoom_start=4)

# show the map
m

In [34]:
folium.Choropleth(
    geo_data=state_geo,
    name="choropleth",
    data=ThirdLineCalledOn,
    columns=["State", "Patients"],
    key_on="feature.id",
    fill_color="YlGn",
    fill_opacity=0.7,
    line_opacity=.1,
    legend_name="3L Patients, Live Rep Interaction, No Qinlock",
).add_to(m)

folium.LayerControl().add_to(m)

m

In [33]:
# "Total 3L+ GIST Patients"
thirdL = df[(df["2Lplus"] == "Yes")
    & (df["Total 3L+ GIST Patients"] > 0)]
thirdL.shape
#thirdL["Total 3L+ GIST Patients"].groupby(df["Facility State"]).count().to_csv("/Users/ganesha/Downloads/TotalHotTargets.csv")
TotalThirdLine = pd.read_csv("/Users/ganesha/Downloads/TotalHotTargets.csv")
TotalThirdLine

State  3LPatients
0     AL           8
1     AR           3
2     AZ           3
3     CA          18
4     CO           2
5     CT           1
6     DC           4
7     DE           1
8     FL          14
9     GA          10
10    HI           6
11    IA           2
12    ID           1
13    IL           6
14    IN           2
15    KS           1
16    LA           4
17    MA           8
18    MD           5
19    MI           6
20    MN           4
21    MO           3
22    MS           3
23    MT           1
24    NC           8
25    NH           1
26    NJ           6
27    NV           2
28    NY          24
29    OH          10
30    OK           1
31    OR           1
32    PA           8
33    PR           1
34    TN           3
35    TX          22
36    UT           3
37    VA           4
38    WA           5
39    WI           6
40    WV           4

In [35]:
# initialize the map and store it in a m object
n = folium.Map(location=[40, -95], zoom_start=4)

folium.Choropleth(
    geo_data=state_geo,
    name="choropleth",
    data=TotalThirdLine,
    columns=["State", "3LPatients"],
    key_on="feature.id",
    fill_color="YlOrRd",
    fill_opacity=0.7,
    line_opacity=.1,
    legend_name="Total 3L Patients",
).add_to(n)

folium.LayerControl().add_to(n)

n